### PR TITLE
[codeql] Address several warnings and errors identified in code scanning

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Chips away at several codeql errors and warnings on the list.  See comments in the commits for problem description and resolution.

## Steps to test these changes
No visible changes should be observed, outside of our warning list going down.

### CODE REVIEW

File: src/map/grades.cpp [hunk 1]
```diff
@@ -191,12 +191,12 @@ namespace grade
         return StatScale[rank][scale];
     }
 
-    float GetMobHPScale(uint8 rank, uint8 scale)
+    uint8 GetMobHPScale(uint8 rank, uint8 scale)
```
Comment (Frankie-hz) [inline src/map/grades.cpp:194]: Ah good catch - my bad!

**Comment by cocosolos** [comment]
The file logging for those migrations was because depending on the state of the database there could be a rather large amount of corrupted data. Those are the only ones that do it because AFAIK they're the only ones that work directly on blob data where each entry has a possibility of being corrupt. For instance if someone made an incorrect backup of their database then every character could potentially have corrupted data there, or if they're importing single characters doing something like a transfer from another server then those characters may have corrupt data and would need to be manually adjusted. I went with a file output for those so they could be reviewed more easily if there's any issue, which may be problematic if relying on printed output, especially if there's a lot of corrupt data possibly resulting in thousands of lines of output depending on server size. However, these migrations would only be run by someone using data from a very old database at this point, or a very screwed up database. I also think dbtool clears the screen at some points when drawing the menu, though I'm not sure if that's still the case.

Sorry for the long write up, but figured I'd explain the rationale there. I don't think I can see any of the warnings/errors you're getting for this stuff unless I'm missing it somewhere.

**Comment by claywar** [comment]
Good point, restored the file logging for these migrations but changed the logic to use `with` to ensure that regardless of exception thrown, the file is closed at the end of each function.

**Review by Xaver-DaRed — APPROVED** [review]

**Review by TeoTwawki — APPROVED** [review]
lgtm

**Pull request merged**